### PR TITLE
Show cocina for version from cocina models button.

### LINKED ISSUE #4496: Display cocina for user version (cocina model button)
Clicking the Cocina model button should open the cocina for that user version in a separate window.

image

### PR DESCRIPTION
**Pull request by justinlittman**
closes #4496

# Why was this change made?

# How was this change tested?

Unit, QA

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -34,6 +34,31 @@ def default_html_head
...
+    message = "Error building Cocina: #{exception.message.truncate(200)}"
+    Honeybadger.notify(exception)
+    logger.error(message)
+    if turbo_frame_request?
+      render 'error', locals: { message: }
+    else
+      redirect_to solr_document_path(params[:item_id] || params[:id]),
+                  flash: { error: message }
+    end
+  end
```
Comment (mjgiarlo) [inline app/controllers/application_controller.rb:60]: This all shows as uncovered. Worth adding coverage to a request spec? (I know this application controller code applies to all controllers equally.)

**Review by mjgiarlo — APPROVED** [review]
Approve pending consideration of testing-related feedback

File: app/models/repository.rb [hunk 2]
```diff
@@ -26,6 +26,20 @@ def self.find_lite(id, structural: true)
...
+  # @return [Cocina::Models::DRO,Cocina::Models::Collection,Cocina::Models::AdminPolicy,NilModel] cocina model instance corresponding to the given druid
+  # @raise [Dor::Services::Client::NotFoundResponse] when dor-services-app cannot find the requested user version
+  def self.find_user_version(id, user_version)
+    raise ArgumentError, 'Missing identifier' unless id
+    raise ArgumentError, 'Missing user_version' unless user_version
+
+    object_client = Dor::Services::Client.object(id)
+    object_client.user_version.find(user_version)
+  rescue Dor::Services::Client::UnexpectedResponse
+    NilModel.new(id)
```
Comment (mjgiarlo) [inline app/models/repository.rb:40]: Also shows as uncovered.

Comment (justinlittman) [reply app/models/repository.rb:40]: So I think NilModel is vestigial, but I'm not entirely sure. I'm going to add an HB alert to help determine.

Comment (mjgiarlo) [reply app/models/repository.rb:40]: Great idea.

**Pull request merged**